### ISSUE TITLE
compression ratio is incorrect if ashift != 9

### ISSUE THREAD

**Issue by ggzengel** [zfsonlinux/zfs#3641]
A compression ration of 2.0 means the data needs half the space?
So it should be 
data_size / used_size = ratio

1.72 / 1.51 = 1.139 but it shows 1.31

```
root@zfs1:~# zfs get all zpool1/samba/imagepool/imagepool2
NAME                               PROPERTY              VALUE                            SOURCE
zpool1/samba/imagepool/imagepool2  type                  filesystem                       -
zpool1/samba/imagepool/imagepool2  creation              Sun Jul 19 15:23 2015            -
zpool1/samba/imagepool/imagepool2  used                  1.51T                            -
zpool1/samba/imagepool/imagepool2  available             33.8T                            -
zpool1/samba/imagepool/imagepool2  referenced            1.51T                            -
zpool1/samba/imagepool/imagepool2  compressratio         1.31x                            -
zpool1/samba/imagepool/imagepool2  mounted               yes                              -
zpool1/samba/imagepool/imagepool2  quota                 none                             default
zpool1/samba/imagepool/imagepool2  reservation           none                             default
zpool1/samba/imagepool/imagepool2  recordsize            128K                             default
zpool1/samba/imagepool/imagepool2  mountpoint            /srv/samba/imagepool/imagepool2  inherited from zpool1/samba
zpool1/samba/imagepool/imagepool2  sharenfs              off                              default
zpool1/samba/imagepool/imagepool2  checksum              on                               default
zpool1/samba/imagepool/imagepool2  compression           lz4                              inherited from zpool1
zpool1/samba/imagepool/imagepool2  atime                 on                               default
zpool1/samba/imagepool/imagepool2  devices               on                               default
zpool1/samba/imagepool/imagepool2  exec                  on                               default
zpool1/samba/imagepool/imagepool2  setuid                on                               default
zpool1/samba/imagepool/imagepool2  readonly              off                              default
zpool1/samba/imagepool/imagepool2  zoned                 off                              default
zpool1/samba/imagepool/imagepool2  snapdir               hidden                           default
zpool1/samba/imagepool/imagepool2  aclinherit            restricted                       default
zpool1/samba/imagepool/imagepool2  canmount              on                               default
zpool1/samba/imagepool/imagepool2  xattr                 on                               default
zpool1/samba/imagepool/imagepool2  copies                1                                default
zpool1/samba/imagepool/imagepool2  version               5                                -
zpool1/samba/imagepool/imagepool2  utf8only              off                              -
zpool1/samba/imagepool/imagepool2  normalization         none                             -
zpool1/samba/imagepool/imagepool2  casesensitivity       sensitive                        -
zpool1/samba/imagepool/imagepool2  vscan                 off                              default
zpool1/samba/imagepool/imagepool2  nbmand                off                              default
zpool1/samba/imagepool/imagepool2  sharesmb              off                              default
zpool1/samba/imagepool/imagepool2  refquota              none                             default
zpool1/samba/imagepool/imagepool2  refreservation        none                             default
zpool1/samba/imagepool/imagepool2  primarycache          all                              default
zpool1/samba/imagepool/imagepool2  secondarycache        all                              default
zpool1/samba/imagepool/imagepool2  usedbysnapshots       0                                -
zpool1/samba/imagepool/imagepool2  usedbydataset         1.51T                            -
zpool1/samba/imagepool/imagepool2  usedbychildren        0                                -
zpool1/samba/imagepool/imagepool2  usedbyrefreservation  0                                -
zpool1/samba/imagepool/imagepool2  logbias               latency                          default
zpool1/samba/imagepool/imagepool2  dedup                 off                              default
zpool1/samba/imagepool/imagepool2  mlslabel              none                             default
zpool1/samba/imagepool/imagepool2  sync                  standard                         default
zpool1/samba/imagepool/imagepool2  refcompressratio      1.31x                            -
zpool1/samba/imagepool/imagepool2  written               1.51T                            -
zpool1/samba/imagepool/imagepool2  logicalused           1.72T                            -
zpool1/samba/imagepool/imagepool2  logicalreferenced     1.72T                            -
zpool1/samba/imagepool/imagepool2  snapdev               hidden                           default
zpool1/samba/imagepool/imagepool2  acltype               off                              default
zpool1/samba/imagepool/imagepool2  context               none                             default
zpool1/samba/imagepool/imagepool2  fscontext             none                             default
zpool1/samba/imagepool/imagepool2  defcontext            none                             default
zpool1/samba/imagepool/imagepool2  rootcontext           none                             default
zpool1/samba/imagepool/imagepool2  relatime              off                              default
zpool1/samba/imagepool/imagepool2  redundant_metadata    all                              default
zpool1/samba/imagepool/imagepool2  overlay               off                              default
```

**Comment by ahrens** [comment]
The compression ratio is not calculated correctly when the ashift of any device in the pool is not 9 (e.g. when using 4K-sector drives).  The problem is that the padding required to align the allocation from 512 -> 4K dilutes the effective compression ratio but is not taken into account.

@ggzengel can you confirm that you are running with 4K drives (ashift=12)?

**Comment by ahrens** [comment]
I just realized that I fixed this a long time ago, in ZoL 0.6.5: c3520e7f1f567bd4e6a28eff4867c70850e8a854
